### PR TITLE
Support pre v3.2 "interfaceRequest"

### PR DESCRIPTION
**Pull request by giovanism**
What I've done:
- Update docs to use new `interface` property
- Implement Unmarshaler that honors `interfaceRequest` alternatif to `interface`

Fix https://github.com/intel/multus-cni/issues/294

### CODE REVIEW

File: types/conf.go [hunk 1]
```diff
@@ -330,3 +330,42 @@ func CheckSystemNamespaces(namespace string, systemNamespaces []string) bool {
 	}
 	return false
 }
+
+// UnmarshalJSON unmarshal Network Selection Annotation. This also supports
+// the deprecated "interfaceRequest" property.
+func (net *NetworkSelectionElement) UnmarshalJSON(bytes []byte) error {
```
Comment (SchSeba) [inline types/conf.go]: where you call this function?

Comment (s1061123) [reply types/conf.go]: UnmarshalJSON() will be invoked in Unmarshal(NetworkSelectionElement).

See followings:
- https://golang.org/pkg/encoding/json/#RawMessage.UnmarshalJSON
- https://mariadesouza.com/2017/09/07/custom-unmarshal-json-in-golang/

Comment (giovanism) [reply types/conf.go]: Thanks for your explanation @s1061123 . Though I'm not really sure if types/conf.go is the right place to write this function.

Comment (nicklesimba) [reply types/conf.go]: I'll look into unit tests for this.

Comment (s1061123) [reply types/conf.go]: @giovanism we may change the function to somewhere in future, but for now, it is okey.

**Pull request merged**